plugin/pulpcore/plugin/changeset/main.py

```diff
@@ -206,11 +212,32 @@ def apply(self):
             {
                 'r': self.repository.name
             })
-        for report in self._apply_additions():
-            yield report
-        for report in self._apply_removals():
+
+        for report in itertools.chain(self._apply_additions(), self._apply_removals()):
+            if report.error:
+                self.failed += 1
+            elif report.action == ChangeReport.ADDED:
+                self.added += 1
+            else:
+                self.removed += 1
             yield report
 
+        log.info(
+            _('ChangeSet complete: added:%(a)d, removed:%(r)d, failed:%(f)d'),
+            {
+                'a': self.added,
+                'r': self.removed,
+                'f': self.failed
+            })
+
+    def apply_and_drain(self):
+        """
+        Call apply() and throw away the ChangeReport objects. This is useful if you don't care about
+        the details and just want to know at the end how many were added, removed, or failed.
```

Comment:
Suggest other wording:

_Call apply() then iterate and discard the change reports.
This is useful when the caller is only interested in the added, removed and failed statistics._